### ISSUE TITLE
Warn/Request User Permission Before Downloading Model

### ISSUE THREAD

**Issue by talaviram** [divamgupta/diffusionbee-stable-diffusion-ui#9]
Currently once dragged, it downloads ~4GB without requesting the user.

Especially on Macs where soldered storage can be even 256GB, it would be nice at least to suggest it's gonna download it to make sure you're connected using a broadband (and not capped cellular or slow network).

**Comment by caunus** [comment]
Yes, this would be helpful. I have a slow connection, but already downloaded the model before, so I would also like to see an option to integrate this already downloaded model instead of downloading again.

**Comment by elbowdonkey** [comment]
DiffusionBee stores the downloaded model as `~/.diffusionbee/downloads/sd-v1-4.cpkt` - perhaps its possible to create that directory before launching DiffusionBee for the first time and copy your desired model to it (which would probably need to be a Stable Diffusion checkpoint), naming the checkpoint as `sd-v1-4.cpkt`? I'd be willing to try swapping out an already existing model and let you know how it goes.

**Comment by kmyqchy** [comment]
> DiffusionBee stores the downloaded model as `~/.diffusionbee/downloads/sd-v1-4.cpkt` - perhaps its possible to create that directory before launching DiffusionBee for the first time and copy your desired model to it (which would probably need to be a Stable Diffusion checkpoint), naming the checkpoint as `sd-v1-4.cpkt`? I'd be willing to try swapping out an already existing model and let you know how it goes.

Its work! Open DiffusionBee and immediately close program (maybe return something wrong) and replaced `~/.diffusionbee/downloads/sd-v1-4.cpkt` with the already downloaded file.

**Comment by duanemoody** [comment]
> DiffusionBee stores the downloaded model as `~/.diffusionbee/downloads/sd-v1-4.cpkt` - perhaps its possible to create that directory before launching DiffusionBee for the first time and copy your desired model to it (which would probably need to be a Stable Diffusion checkpoint), naming the checkpoint as `sd-v1-4.cpkt`? I'd be willing to try swapping out an already existing model and let you know how it goes.

At least on my machine (Intel) the model is stored as `~/.diffusionbee/downloads/diffusion_model.h5` not `.ckpt`, I can't tell why this is.
